pkg/doc/did/doc.go

```diff
@@ -360,12 +360,61 @@ func ParseDocument(data []byte) (*Doc, error) {
 	} else if raw == nil {
 		return nil, errors.New("document payload is not provided")
 	}
-	// validate did document
-	err = validate(data, raw.schemaLoader())
+
+	switch {
+	case strings.HasPrefix(raw.ID, "did:sov:"):
+		return legacyParseSov(raw)
+	default:
+		// validate did document
+		err = validate(data, raw.schemaLoader())
+		if err != nil {
+			return nil, err
+		}
+
+		return parseDocument(raw)
+	}
+}
+
+func parseDocument(raw *rawDoc) (*Doc, error) {
+	doc := &Doc{
+		ID:      raw.ID,
+		Created: raw.Created,
+		Updated: raw.Updated,
+	}
+
+	context, baseURI := parseContext(raw.Context)
+	doc.Context = context
+	doc.processingMeta = processingMeta{baseURI: baseURI}
+	doc.Service = populateServices(raw.ID, baseURI, raw.Service)
+
+	verificationMethod := raw.PublicKey
+	if len(raw.VerificationMethod) != 0 {
+		verificationMethod = raw.VerificationMethod
+	}
+
+	vm, err := populateVerificationMethod(context[0], doc.ID, baseURI, verificationMethod)
+	if err != nil {
+		return nil, fmt.Errorf("populate verification method failed: %w", err)
+	}
+
+	doc.VerificationMethod = vm
+
+	err = populateVerificationRelationships(doc, raw, false)
 	if err != nil {
 		return nil, err
 	}
 
+	proofs, err := populateProofs(context[0], doc.ID, baseURI, raw.Proof)
+	if err != nil {
+		return nil, fmt.Errorf("populate proofs failed: %w", err)
+	}
+
+	doc.Proof = proofs
+
+	return doc, nil
+}
+
+func legacyParseSov(raw *rawDoc) (*Doc, error) {
```

Comment:
@swcurran @andrewwhitehead DID documents now use the verificationMethod property not publicKey. We are getting an invalid DID document.

Reply:
Again need to check with the .NET framework but I don't think it will accept verificationMethod.

Reply:
@andrewwhitehead they will need to fix this. https://www.w3.org/TR/did-core/#verification-methods

Reply:
It's about a community upgrade. When the first cut was implemented "publicKeys" was the attribute name in the then DID Core spec.

Reply:
Yup. Created https://github.com/hyperledger/aries-cloudagent-python/issues/1050